synapse/handlers/sync.py

```diff
@@ -1817,11 +1817,34 @@ async def _generate_sync_entry_for_rooms(
             )
             sync_result_builder.now_token = now_token
 
+        # Retrieve rooms that got un partial stated in the meantime, only useful in case
+        # of a non lazy-loading-members sync.
+        un_partial_stated_rooms = set()
+        if not sync_result_builder.sync_config.filter_collection.lazy_load_members():
+            un_partial_state_rooms_since = 0
+            if sync_result_builder.since_token is not None:
+                un_partial_state_rooms_since = int(
+                    sync_result_builder.since_token.un_partial_state_rooms_key
+                )
+
+            un_partial_state_rooms_now = int(
+                sync_result_builder.now_token.un_partial_state_rooms_key
+            )
```

Comment:
In both cases, isn't `un_partial_state_rooms_key` already an int?

Reply:
Indeed, it was inspired by `to_device_key` and `push_rules_key` that has similar useless cast. It probably predates mypy use and is legacy.

I'll remove them.